specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json

```diff
@@ -8263,6 +8507,198 @@
+        "resource"
+      ]
+    },
+    "RedistributeThroughputPropertiesResource": {
+      "type": "object",
+      "description": "Resource to redistribute throughput for Azure Cosmos DB resource",
+      "properties": {
+        "throughputPolicy": {
+          "description": "ThroughputPolicy to apply for throughput redistribution",
+          "type": "string",
+          "enum": [
+            "none",
+            "equal",
+            "custom"
+          ],
+          "x-ms-enum": {
+            "name": "ThroughputPolicyType",
+            "modelAsString": true
+          }
+        },
+        "targetPhysicalPartitionThroughputInfo": {
+          "type": "array",
+          "description": "Array of PhysicalPartitionThroughputInfoResource objects.",
+          "items": {
+            "$ref": "#/definitions/PhysicalPartitionThroughputInfoResource"
+          }
+        },
+        "sourcePhysicalPartitionThroughputInfo": {
+          "type": "array",
+          "description": "Array of PhysicalPartitionThroughputInfoResource objects.",
+          "items": {
+            "$ref": "#/definitions/PhysicalPartitionThroughputInfoResource"
+          }
+        }
+      },
+      "required": [
+        "throughputPolicy",
+        "targetPhysicalPartitionThroughputInfo",
+        "sourcePhysicalPartitionThroughputInfo"
+      ]
+    },
+    "RetrieveThroughputParameters": {
+      "description": "Cosmos DB retrieve throughput parameters object",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to retrieve throughput parameters object",
+          "$ref": "#/definitions/RetrieveThroughputProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ARMResourceProperties"
+        }
+      ],
+      "required": [
+        "properties"
+      ]
+    },
+    "RetrieveThroughputProperties": {
+      "description": "Properties to retrieve throughput for Azure Cosmos DB resource.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The standard JSON format of a resource throughput",
+          "$ref": "#/definitions/RetrieveThroughputPropertiesResource"
+        }
+      },
+      "required": [
+        "resource"
+      ]
+    },
+    "RetrieveThroughputPropertiesResource": {
+      "type": "object",
+      "description": "Resource to retrieve throughput information for Cosmos DB resource",
+      "properties": {
+        "physicalPartitionIds": {
+          "type": "array",
+          "description": "Array of PhysicalPartitionId objects.",
+          "items": {
+            "$ref": "#/definitions/PhysicalPartitionId"
+          }
+        }
+      },
+      "required": [
+        "physicalPartitionIds"
+      ]
+    },
+    "PhysicalPartitionId": {
+      "type": "object",
+      "description": "PhysicalPartitionId object",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

What kind of id is this? Is it specific to this RP, or an Azure resource ID? A name, guid, something else? It is helpful if the API gives clues about the nature of the Id in the name, description, or both.

Reply:
It is id of a physical partition within cosmosdb. Cosmosdb supports offerings at collection and at database level, meaning a single physical partition can host single collection or multiple collections. Depending on which one it is, this id can be a guid or a non-negative integer. Since partitioning is at the core of cosmosdb and is referred to throughout it's documentation, I don't think it needs any elaboration here.